### ISSUE TITLE
Bug: [ViewCells] private properties do not work with Computed Properties

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#7459]
### PHP Version

8.1

### CodeIgniter4 Version

4.3.3 and develop fe7e2e497075073f11c6b36c8d8dac070ea595f8

### CodeIgniter4 Installation Method

Git

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### What happened?

See https://github.com/codeigniter4/CodeIgniter4/pull/7457#discussion_r1178771926
```
Cannot access private property App\Cells\AlertMessageCell::$type
in SYSTEMPATH/Traits/PropertiesTrait.php on line 35.
 1 SYSTEMPATH/View/Cell.php(214): CodeIgniter\View\Cells\Cell->fill([...])
 2 SYSTEMPATH/View/Cell.php(99): CodeIgniter\View\Cell->renderCell(Object(App\Cells\AlertMessageCell), 'render', [...])
 3 SYSTEMPATH/Common.php(1200): CodeIgniter\View\Cell->render('AlertMessageCell', [...], 0, 'AppCellsAlertMessageCellrender3ab314af93d2044b8409e6348dd9ff46')
```

### Steps to Reproduce

```diff
--- a/tests/_support/View/Cells/ListerCell.php
+++ b/tests/_support/View/Cells/ListerCell.php
@@ -15,7 +15,7 @@ use CodeIgniter\View\Cells\Cell;
 
 class ListerCell extends Cell
 {
-    protected array $items = [];
+    private array $items = [];
 
     public function getItemsProperty()
     {
```

### Expected Output

No error.

**Comment by michalsn** [comment]
It seems like everything is good. The description for `protected` and `private` properties refers to the [computed properties](https://www.codeigniter.com/user_guide/outgoing/view_cells.html#computed-properties). The example you're showing here is something different:

```php
    public function testCellWithComputedProperties()
    {
        $result = view_cell(ListerCell::class, ['items' => ['one', 'two', 'three']]);
```

The `$items` are added during cell initialization. This is why we are getting an error. We could never initialize properties that were `private`.

---

This would be a good example with using the `private` computed property:

```php
// initializing
view_cell('AlertMessageCell', ['type' => 'note', 'message' => 'test']);

// cell
class AlertMessageCell extends Cell
{
    protected $type;
    protected $message;
    private $computed;

    public function mount()
    {
        $this->computed = sprintf('%s - %s', $this->type, $this->message);
    }

    public function getComputedProperty(): string
    {
        return $this->computed;
    }

    public function getTypeProperty(): string
    {
        return $this->type;
    }

    public function getMessageProperty(): string
    {
        return $this->message;
    }
}

// view
<div>
    <p>type - <?= $type; ?></p>
    <p>message - <?= $message; ?></p>
    <p>computed: <?= $computed; ?></p>
</div>
```

**Comment by kenjis** [comment]
You mean, we cannot pass the `private` computed properties in `view_cell()`?
If such is the specification, then the difference between `protected` and `private` makes sense.

**Comment by michalsn** [comment]
Yes, this is intended. We can't set properties that are declared as `private` during cell initialization. We can't allow passing values to every property because this would enable to override of any variable we defined.

**Comment by kenjis** [comment]
Okay, I send a PR #7460

**Comment by kenjis** [comment]
Closed by #7460
